zrml/prediction-markets/src/lib.rs

```diff
@@ -328,6 +328,11 @@ mod pallet {
                 categories <= T::MaxCategories::get(),
                 <Error<T>>::TooManyCategories
             );
+            // Require sha3-384 as multihash.
+            ensure!(
+                metadata[0] == 0x15 && metadata[1] == 0x30,
+                <Error<T>>::InvalidMultihash
+            );
```

Comment:
Our chain filters out any calls that contain metadata, which does not use the multihash identifier byte and length byte for sha3-384.

This ensures that all calls at least have the correct identifiers, potentially serving as a protection against invalid calls beforehand and therefore market creation with invalid metadata. It also enforces the hash function to sha3-384, denying any call that supplies metadata which does not use the corresponding identifiers. For anybody who just wants to play around locally, this may not be trivial, but this can be improved by adding nice docstrings (#119). Alternatively, we can increase the static size of the metadata array by 3 bytes and removing the checks, effectively allowing any hash function with up to 48 bytes digest size to be used.

What do you prefer? My position is we think about which data types we want once and then enforce it, which leads to uniformity and less error-prone usage (in case someone who uses it knows how it should be used - docs).

Reply:
> For anybody who just wants to play around locally, this may not be trivial, but this can be improved by adding nice docstrings (#119).

A couple things we could do here,

1) make `Market.metadata` an `Option` so that it's possible to submit `None`. But if it is provided it must be a well-formatted multihash. 
2) make `Market.metadata` an new enum, that can extended over time:

```rust
enum Metadata {
  MultihashSha3_384([u8;50]),
  <new formats we want to support in the future>
  None,
}
```

Reply:
> My position is we think about which data types we want once and then enforce it, which leads to uniformity and less error-prone usage (in case someone who uses it knows how it should be used - docs).

I agree, if we do support new variants for metadata in the future, they should always be well specified and enforced in chain. In an ideal world, we would also enforce the JSON is correct with the expected fields. I see this possibly being the case if instead of adding to IPFS and submitting to Zeitgeist in two steps, we added the content to IPFS from the runtime, but I would leave this for a later improvement and pending further discussion.

Reply:
@lsaether So no metadata at all is also a valid option?

Reply:
Yeah, I think it would be okay, the market would be pretty useless but it's at the cost of the creator. Also it would aid in testing to not need to have a local tool handy to create the multihash.